torchtitan/profiling.py

```diff
@@ -27,8 +27,7 @@ def maybe_enable_profiling(config: JobConfig, *pos_args, **kwargs):
         trace_dir = os.path.join(dump_dir, save_trace_dir)
         profile_freq = config.profiling.profile_freq
 
-        _global_iter_count = 0
-
+        _global_iter_count = global_step
```

Comment:
This does not sync with the internal state `step_num` of profiler. Let's remove it and replace all appearance of `_global_iter_count` by `prof.step_num` in `trace_handler`.

Also need to set `torch_profiler.step_num = global_step` on line 71.